authors/apps/authentication/serializers.py

```diff
@@ -25,6 +30,51 @@ class Meta:
         # or response, including fields specified explicitly above.
         fields = ['email', 'username', 'password']
 
+    def validate_email(self, value):
+
+        email_regex = (r'^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$')
+        data = self.get_initial()
+        email = data.get("email")
+        user_query_set = User.objects.filter(email=email)
+
+        # validate that the correct email format is used
+        if not re.search(email_regex, email):
+            raise serializers.ValidationError(
+                "Incorrect email format please try again")
+        return value
+
+        if user_query_set.exists():
+            raise serializers.ValidationError(
+                "This email has already been used to create a user")
+        return value
+
+    def validate_username(self, value):
+
+        username_regex = (r'^[a-zA-Z]+$')
+        data = self.get_initial()
+        username = data.get("username")
+        user_query_set = User.objects.filter(username=username)
+
+        # validate that username passed is correct
+        if not re.search(username_regex, username):
+            raise serializers.ValidationError(
+                "The username is incorrect please use characters")
```

Comment:
hey @wasibani-roy, from the error message ``` "The username is incorrect please use characters"``` a character can mean anything (a letter,a number,punctualtion marks etc) [Definition of a character](https://en.wikipedia.org/wiki/Character_(computing)")
from the regex check i think we could write the error as 
> The username is invalid, it should contain only letters,

Reply:
Still not very sure about this please throw more light